mavis/validate/call.py

```diff
@@ -753,12 +813,40 @@ def _call_by_supporting_reads(evidence, event_type, consumed_evidence=None):
             if deletion_size > evidence.max_expected_fragment_size and event_type == SVTYPE.INS:
                 continue
 
-        first_breakpoint = Breakpoint(evidence.break1.chr, first, strand=evidence.break1.strand, orient=evidence.break1.orient)
-        second_breakpoint = Breakpoint(evidence.break2.chr, second, strand=evidence.break2.strand, orient=evidence.break2.orient)
-        call = EventCall(
-            first_breakpoint, second_breakpoint, evidence, event_type,
-            call_method=CALL_METHOD.SPLIT
-        )
+        # check if any of the aligned reads are 'double' aligned
+        double_aligned = {}
+        for read in pos1[first] | pos2[second]:
+            seq_key = tuple(sorted([
+                read.query_name, read.query_sequence, reverse_complement(read.query_sequence)
+            ]))  # seq and revseq are equal
+            double_aligned.setdefault(seq_key, []).append(read)
+
+        # now create calls using the double aligned split read pairs if possible (to resolve untemplated sequence)
+        resolved_calls = set()
+        event_types = {event_type}
+        if event_type in {SVTYPE.DUP, SVTYPE.INS}:
+            event_types.update({SVTYPE.DUP, SVTYPE.INS})
+        for reads in [d for d in double_aligned.values() if len(d) > 1]:
+            for read1, read2 in itertools.combinations(reads, 2):
+                try:
+                    call = call_paired_read_event(read1, read2)
+                    if BreakpointPair.classify(call) & event_types:  # ensure we are calling the correct event types
+                        resolved_calls.add(call)
+                except AssertionError:
```

Comment:
homologous recombination defeciency

Reply:
ya I can add a break here when resolved_calls > 1. Shouldn't make too much of a difference speed-wise since there will generally be very few combinations to go through